nominal/core.py

```diff
@@ -311,6 +339,132 @@ def _from_conjure(cls, client: NominalClient, dataset: scout_catalog.EnrichedDat
         )
 
 
+@dataclass(frozen=True)
+class LogSet:
+    rid: str
+    created_by: str
+    name: str
+    created_at: datetime
+    updated_at: datetime
+    log_count: int
+    timestamp_type: TimestampType
+    origin_metadata: Mapping[str, str]
+    description: str | None
+    _client: NominalClient = field(repr=False)
+
+    def add_logs_and_finalize(self, logs: list[Log]) -> Self:
+        """
+        Attaches a list of logs to this log set and finalizes it. A logset is not considered readable
+        until it has been finalized. Once finalized, the logset is immutable.
+        """
+        request = datasource_logset_api.AttachLogsAndFinalizeRequest(
+            logs=[log._to_conjure() for log in logs],
+        )
+        response = self._client._logset_client.attach_logs_and_finalize(
+            auth_header=self._client._auth_header, log_set_rid=self.rid, request=request
+        )
+        log_set_metadata = self.__class__._from_conjure(self._client, response)
+        update_dataclass(self, log_set_metadata, fields=self.__dataclass_fields__)
+        return self
+
+    def get_logs_paginated(
+        self,
+        page_size: int | None = None,
+    ) -> PaginatedLogs:
+        """
+        Get logs within this log set. Paginated, maximum page size is 10000, defauls to maximum.
+        """
+        response = self._client._logset_client.search_logs(
+            self._client._auth_header,
+            log_set_rid=self.rid,
+            request=datasource_logset_api.SearchLogsRequest(page_size=page_size),
+        )
+        return PaginatedLogs(
+            logs=[Log._from_conjure(log) for log in response.logs],
+            page_size=page_size,
+            next_page_token=response.next_page_token,
+            client=self._client,
+        )
```

Comment:
Don't love the workflow here, of like `get_log_set().get_logs_paginated().get_logs()`